### PR TITLE
SONARJAVA-2055 Enable locations for XML element in case of IBM JVMs.

### PR DESCRIPTION
**Pull request by m-g-sonar**
I actually have no idea how to test it except from manual tests. I reproduced the issue with an IBM jdk and sonalint CLI, while trying to analyze any POM file.

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/xml/maven/PomParser.java [hunk 1]
```diff
@@ -68,6 +69,22 @@ public static MavenProject parseXML(File file) {
...
+   * The property is a IBM-specific property, not recognized by non-IBM JVMs, and will throw an {@link IllegalArgumentException}.
+   *
+   * See {@link https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.zos.80.doc/user/xml/xlxpj_reference.html}.
+   *
+   * @param factory
+   */
+  private static void enableLocationPropertyForIBM(XMLInputFactory factory) {
+    try {
+      factory.setProperty("javax.xml.stream.isSupportingLocationCoordinates", true);
+    } catch (IllegalArgumentException e) {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/xml/maven/PomParser.java]: MAJOR Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: java-frontend/src/main/java/org/sonar/java/xml/maven/PomParser.java [hunk 2]
```diff
@@ -68,6 +69,22 @@ public static MavenProject parseXML(File file) {
...
+  /**
+   * By default, the location of XML element is enabled, except on IBM JVM where it is disabled and has to be manually enabled.
+   * The property is a IBM-specific property, not recognized by non-IBM JVMs, and will throw an {@link IllegalArgumentException}.
+   *
+   * See {@link https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.zos.80.doc/user/xml/xlxpj_reference.html}.
+   *
+   * @param factory
+   */
+  private static void enableLocationPropertyForIBM(XMLInputFactory factory) {
+    try {
```
Comment (saberduck) [inline java-frontend/src/main/java/org/sonar/java/xml/maven/PomParser.java]: You should be able to use `isPropertySupported` API on `XMLInputFactory` ,  that way you can perhaps avoid try-catch statement. But I am not sure if it works with IBM JDK, I just saw it in the javadoc.

Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/xml/maven/PomParser.java]: nice! missed that one! Thanks! It worth trying!

**Pull request merged**